kong/db/schema/init.lua

```diff
@@ -1704,6 +1706,30 @@ local function collect_field_reference(refs, key, reference)
 end
 
 
+local function validate_deprecation_exclusiveness(data, shorthand_value, shorthand_name, shorthand_definition)
+  if shorthand_value and shorthand_value ~= ngx.null and shorthand_definition.deprecation and shorthand_definition.deprecation.replaced_with then
```

Comment:
nit: reverse the logic and return early. makes it easier to read due to less indentation, but feel free to have that as a follow-up fix if you want.

Reply:
done ✅